src/dackar/utils/mbse/customMBSEparser.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2024, Battelle Energy Alliance, LLC  ALL RIGHTS RESERVED
+
+"""
+Created on February, 2024
+
+@author: mandd
+"""
+
+# External Imports
+import pandas as pd
+import logging
+
+logger = logging.getLogger("my logger")
+c_handler = logging.StreamHandler()
+logger.addHandler(c_handler)
+logger.setLevel(logging.DEBUG) 
+
+class customMBSEobject(object):
+    """
+        Class designed to process the a custom MBSE model from file.
+    """
+    def __init__(self, nodes_filename, edges_filename):
+        """
+        Initialization method for the custom MBSE model class
+
+        Args:
+
+            nodes_filename: file, file in .csv format containing all nodes 
+            edges_filename: file, file in .csv format containing all edges 
+
+        Returns:
+
+            None
+        """
+        self.nodes_filename = nodes_filename
+        self.edges_filename = edges_filename
+        self.list_IDs = []
+
+        self.allowed_node_types = ['entity']
+        self.allowed_edge_types = ['link','composition','support'] # to be developed: 'opm_instance'
+
+        self.allowed_node_cols = ['label','ID','type']
+        self.allowed_edge_cols = ['sourceNodeId','targetNodeId','type','medium']
+
+        self.parseFiles()
+        self.checkNodes()
+        self.checkEdges()
+
+        nodes_file_split =  self.nodes_filename.split('.')
+        nodes_file_kg = nodes_file_split[0] + '_kg.' + nodes_file_split[1]
+
+        edges_file_split =  self.edges_filename.split('.')
+        edges_file_kg = edges_file_split[0] + '_kg.' + edges_file_split[1]
+
+        self.printOnFiles(nodes_file_kg,edges_file_kg)
+    
+    def checkModel(self):
+        """
+        Method designed to pcheck model consistency
+
+        Args:
+
+            None
+
+        Returns:
+
+            None
+        """
+        self.checkNodes()
+        self.checkEdges()       
+    
+    def parseFiles(self):
+        """
+        Method designed to parse the node and edge files
+
+        Args:
+
+            None
+
+        Returns:
+
+            None
+        """
+        # parse nodes
+        self.nodes_df = pd.read_csv(self.nodes_filename, sep=',', skip_blank_lines=True, dtype=str)
+        self.nodes_df.dropna(how='all', inplace=True)
+        self.nodes_df = self.nodes_df.apply(lambda x: x.astype(str).str.lower())
+
+        self.list_IDs = self.nodes_df['ID'].dropna().to_list()
+        
+        # parse edges
+        self.edges_df = pd.read_csv(self.edges_filename, sep=',', skip_blank_lines=True, dtype=str)
+        self.edges_df.dropna(how='all', inplace=True)
+        self.edges_df = self.edges_df.apply(lambda x: x.astype(str).str.lower())
```

Comment:
There are a couple of variable with underscores, I may suggest to change them to camelBack format.